common/leica_sep.c

```diff
@@ -63,27 +104,51 @@ static int leica_sep_transfer(struct leica_sep_funcs *f, char *cmd, char *data,
 	if (!command_received)
 		return -1;
 
+	if(leica_get_sep_payload(data, bytes_received))
```

Comment:
We iterate over all characters received, put all characters in a buffer and detect the start of the payload (to validate correct reception).
So everything is just prepared to fill the buffer with payload only...
But we iterate again over everything received, do again a detection of payload and modify the buffer we are reading.
We not just do it in this function? I don't see a possible scenario, where this function could be reused.